safe-global/safe-wallet-web
# feat: Add MetaMask SDK to onboard.js

## What it solves

Resolves #3841

## How this PR fixes it

- Adds the MetaMask SDK as a wallet module to onboard.js

## How to test it

1. Open Safe and try to connect with MetaMask
2. The MM extension should open normally

- Test what happens when no MM extension is installed
- Test what happens on mobile with MM app installed and without
-> iOS: If no MM app is installed it opens the app store page of MM
-> iOS: If MM app is installed it opens it and connects

## Screenshots

No extension installed -> User sees the mobile QR code to scan and a dialog to download the extension
<img width="820" alt="Screenshot 2024-07-15 at 12 04 47" src="https://github.com/user-attachments/assets/3ec6a095-02f5-4f5d-b919-3ae760032332">
<img width="817" alt="Screenshot 2024-07-15 at 12 04 51" src="https://github.com/user-attachments/assets/76bfa5d5-2cd6-46b3-926e-bdd1c7233524">



## Checklist
